modules/javafx.swing/src/test/java/test/javafx/embed/swing/JFXPanelTest.java

```diff
@@ -0,0 +1,131 @@
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+
+public class JFXPanelTest {
+
+    // Used to launch the application before running any test
+    private static final CountDownLatch launchLatch = new CountDownLatch(1);
+    private static boolean setupIsDone = false;
+
+    @BeforeClass
+    public static void doSetupOnce() {
+        if (setupIsDone) return;
+        Platform.startup(() -> {
+            Platform.setImplicitExit(false);
+            launchLatch.countDown();
+        });
+
+        try {
+            if (!launchLatch.await(5000, TimeUnit.MILLISECONDS)) {
+                throw new AssertionFailedError("Timeout waiting for Application to launch");
+            }
+        } catch (InterruptedException ex) {
+            AssertionFailedError err = new AssertionFailedError("Unexpected exception");
+            err.initCause(ex);
+            throw err;
+        }
+
+        assertEquals(0, launchLatch.getCount());
+        setupIsDone = true;
+    }
+
+    static class TestFXPanel extends JFXPanel {
+        protected void processMouseEventPublic(MouseEvent e) {
+            processMouseEvent(e);
+        }
+    };
+
+    @Test
+    public void testNoDoubleClickOnFirstClick() throws Exception {
+
+        final CountDownLatch firstPressedEventLatch = new CountDownLatch(1);
+
+        // It's an array, so we can mutate it inside of lambda statement
+        final int[] pressedEventCounter = {0};
+
+        SwingUtilities.invokeLater(() -> {
+            TestFXPanel fxPnl = new TestFXPanel();
+            fxPnl.setPreferredSize(new Dimension(100, 100));
+            JFrame jframe = new JFrame();
+            JPanel jpanel = new JPanel();
+            jpanel.add(fxPnl);
+            jframe.setContentPane(jpanel);
+            jframe.setVisible(true);
+
+            Platform.runLater(() -> {
+                Group grp = new Group();
+                Scene scene = new Scene(new Group());
+                scene.getRoot().requestFocus();
+
+                scene.addEventHandler(javafx.scene.input.MouseEvent.MOUSE_PRESSED, (event -> {
+                    pressedEventCounter[0] += 1;
+                    firstPressedEventLatch.countDown();
+                }));
+
+                fxPnl.setScene(scene);
+
+                SwingUtilities.invokeLater(() -> {
+                    MouseEvent e = new MouseEvent(fxPnl, MouseEvent.MOUSE_PRESSED, 0, MouseEvent.BUTTON1_DOWN_MASK,
+                            5, 5, 1, false, MouseEvent.BUTTON1);
+
+                    fxPnl.processMouseEventPublic(e);
+                });
+            });
+        });
+
+        if (!firstPressedEventLatch.await(5000, TimeUnit.MILLISECONDS)) {
+            throw new Exception("No first click detected.");
+        };
+
+        Thread.sleep(100); // there should be no pressed event after the initial one. Let's wait for 0.1s and check again.
+        assertEquals(1, pressedEventCounter[0]);
+    }
```

Comment:
You will want to add a cleanup method, annotated with `@AfterClass` to call `Platform.exit` and dispose of the JFrame.

Reply:
I've kept the logic of the other JFXPanelTest. Should i change it?